main.go

```diff
@@ -0,0 +1,448 @@
+	if !a.GUI {
+		return addWordsToJSONCLI(a.JsonFile)
+	}
+
+	app := app.New()
+	win := app.NewWindow("A fake drawer in Brno")
+	if err := addWordsToJSONGUI(a.JsonFile, win); err != nil {
+		return err
+	}
+	win.ShowAndRun()
+	return nil
+}
+
+func loadJsonFile(path string) (Words, error) {
+	var w Words
+
+	f, err := os.ReadFile(path)
+	if err != nil {
+		return w, err
+	}
+
+	if err := json.Unmarshal(f, &w); err != nil {
+		return w, err
+	}
+	return w, nil
+}
+
+func saveToJsonFile(path string, w Words) error {
+	input, err := json.Marshal(w)
+	if err != nil {
+		return err
+	}
+
+	if err := os.WriteFile(path, input, 0o660); err != nil {
+		return err
+	}
+	return nil
+}
+
+func loadCategories(w Words) []string {
+	cats := make(map[string]bool)
+	for i := range w {
+		cats[w[i].Category] = true
+	}
+
+	result := make([]string, len(cats))
+	i := 0
+	for k := range cats {
+		result[i] = k
+		i++
+	}
+	return result
+}
+
+func addWordsToJSONGUI(path string, win fyne.Window) error {
+	w, err := loadJsonFile(path)
+	if err != nil {
+		return err
+	}
+
+	categories := widget.NewLabel(fmt.Sprintf("Categories in inputed JSON file: %s", loadCategories(w)))
+	addWordQuestion := widget.NewLabel("Do you want to add another word?")
+	wordEntry := widget.NewEntry()
+	catEntry := widget.NewEntry()
+	form := widget.NewForm(
+		&widget.FormItem{Text: "Word", Widget: wordEntry},
+		&widget.FormItem{Text: "Category", Widget: catEntry})
+	winContainer := container.NewVBox()
+	entryContainer := container.NewVBox()
+
+	confirmationContainer := container.NewVBox(
+		addWordQuestion,
+		container.NewHBox(
+			widget.NewButton("Yes", func() {
+				winContainer.RemoveAll()
+				winContainer.Add(entryContainer)
+			}),
+			widget.NewButton("No", func() {
+				winContainer.RemoveAll()
+				if err = saveToJsonFile(path, w); err != nil {
+					winContainer.Add(widget.NewLabel("Failed to save JSON file!"))
+					winContainer.Add(widget.NewLabel(err.Error()))
+				} else {
+					winContainer.Add(widget.NewLabel("Successfully saved words to JSON file!"))
+				}
+				winContainer.Add(widget.NewButton("Close", win.Close))
+			})))
+
+	form.OnSubmit = func() {
+		if strings.TrimSpace(catEntry.Text) == "" || strings.TrimSpace(wordEntry.Text) == "" {
+			e := dialog.NewError(errors.New("entry boxes cannot be empty"), win)
+			e.Show()
+		} else {
+			w = append(w,
+				Word{
+					Category: strings.TrimSpace(catEntry.Text),
+					Word:     strings.TrimSpace(wordEntry.Text)})
+			catEntry.SetText("")
+			wordEntry.SetText("")
+			winContainer.RemoveAll()
+			winContainer.Add(confirmationContainer)
+		}
+	}
+	form.OnCancel = win.Close
+
+	entryContainer = container.NewVBox(categories, form)
+
+	winContainer.Add(entryContainer)
+
+	win.SetContent(winContainer)
+
+	return err
+}
+
+func addWordsToJSONCLI(path string) error {
+	w, err := loadJsonFile(path)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Categories in inputed JSON file: %s\n", loadCategories(w))
```

Comment:
I would opt for some king of a logger. Either use the default and simple `log`, structured logger `slog` or external library like `zerolog`.